README.md

```diff
@@ -1,52 +1,207 @@
 
-Then you can watch the daemonset with `kubectl -n openshift-cluster-network-operator logs <PODID>`.
+Currently, changing the address pools once set is not supported. In the future, some network providers may support expanding the address pools.
 
-## Example configuration
+Each ClusterNetwork entry has an additional required parameter, `hostSubnetLength`, that specifies the address size to assign to assign to each individual node. Note that this is currently *reverse* from the usual CIDR notation - a hostSubnetLength of 9 means that the node will be assigned a /23.
 
+Example
 ```yaml
-apiVersion: "networkoperator.openshift.io/v1"
-kind: "NetworkConfig"
-metadata:
-  name: "default"
-  namespace: cluster-network-operator
 spec:
   serviceNetwork: "172.30.0.0/16"
   clusterNetworks:
     - cidr: "10.128.0.0/14"
       hostSubnetLength: 9
+    - cidr: "192.168.0.0/18"
+      hostSubnetLength: 9
+```
+
+## Configuring the default network provider
+Users must select a default network provider. This cannot be changed. Different network providers have additional provider-specific settings.
+
+Currently, the only supported value for network Type is `OpenShiftSDN`.
+
+### Configuring OpenShiftSDN
+OpenShiftSDN supports the following configuration options, all of which are optional:
+* `mode`: one of "Subnet" "Multitenant", or "NetworkPolicy". Configures the [isolation mode](https://docs.openshift.com/container-platform/3.11/architecture/networking/sdn.html#overview) for OpenShift SDN. The default is "NetworkPolicy".
+* `vxlanPort`: The port to use for the VXLAN overlay. The default is 4789
+* `MTU`: The MTU to use for the VXLAN overlay. The default is the MTU of the node that the cluster-network-operator is first run on, minus 50 bytes for overhead. If the nodes in your cluster don't all have the same MTU then you will need to set this explicitly.
+* `useExternalOpenvswitch`: boolean. If the nodes are already running openvswitch, and OpenShiftSDN should not install its own, set this to true. This only needed for certain advanced installations with DPDK or OpenStack.
+
+Example:
+```yaml
+spec:
   defaultNetwork:
     type: OpenShiftSDN
     openshiftSDNConfig:
       mode: NetworkPolicy
+      vxlanPort: 4789
+      mtu: 1450
+      useExternalOpenvswitch: false
+```
+
+## Configuring kube-proxy
+Users may customize the kube-proxy configuration. None of these settings are required:
+
+* `iptablesSyncPeriod`: The interval between periodic iptables refreshes. Default: 30 seconds. Increasing this can reduce the number of iptables invocations.
+* `bindAddress`: The address to "bind" to - the address for which traffic will be redirected.
+* `proxyArguments`: additional command-line flags to pass to kube-proxy - see the [documentation](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/).
```

Comment:
should clarify how this works; `proxyArguments` is a `map[string][]string`, so the value has to be specified as an array, even though most arguments only accept a single value. (This might make more sense as a footnote to the example below.)